DevEdu.API/DevEdu.API/Controllers/LessonController.cs

```diff
@@ -49,28 +58,67 @@ public void UpdateLesson(int id, [FromBody] LessonUpdateInputModel updateModel)
 
         // api/lesson/{id}
         [HttpGet("{id}")]
-        public LessonDto GetLessonById(int id)
+        [Description("Get the lesson by id.")]
+        [ProducesResponseType(typeof(LessonInfoOutputModel), StatusCodes.Status200OK)]
+        public LessonInfoOutputModel GetLessonById(int id)
         {
-            return _lessonService.SelectLessonById(id);
+            var dto = _lessonService.SelectLessonById(id);
+            return _mapper.Map<LessonInfoOutputModel>(dto);
         }
 
         // api/lesson
         [HttpGet]
-        public List<LessonDto> GetAllLessons()
+        [Description("Get all lessons.")]
+        [ProducesResponseType(typeof(LessonInfoOutputModel), StatusCodes.Status200OK)]
+        public List<LessonInfoOutputModel> GetAllLessons()
         {
-            return _lessonService.SelectAllLessons();
+            var dto = _lessonService.SelectAllLessons();
+            return  _mapper.Map<List<LessonInfoOutputModel>>(dto);
         }
 
+        // api/lesson/with-topics
+        [HttpGet("with-topics")]
+        [Description("Get all lessons with topics.")]
+        [ProducesResponseType(typeof(LessonInfoWithTopicsOutputModel), StatusCodes.Status200OK)]
+        public List<LessonInfoWithTopicsOutputModel> GetAllLessonsWithTopics()
+        {
+            var dto = _lessonService.SelectAllLessons();
+            return _mapper.Map<List<LessonInfoWithTopicsOutputModel>>(dto);
+        }
+
+        // api/lesson/with-groups
+        [HttpGet("with-groups")]
+        [Description("Get all lessons with groups.")]
+        [ProducesResponseType(typeof(LessonInfoWithGroupsOutputModel), StatusCodes.Status200OK)]
+        public List<LessonInfoWithGroupsOutputModel> GetAllLessonsWithGroups()
+        {
+            var dto = _lessonService.SelectAllLessons();
+            return _mapper.Map<List<LessonInfoWithGroupsOutputModel>>(dto);
+        }
+
+        // api/lesson/with-students-comments
+        [HttpGet("with-students-comments")]
+        [Description("Get all lessons with students and comments.")]
+        [ProducesResponseType(typeof(LessonInfoWithStudentsAndCommentsOutputModel), StatusCodes.Status200OK)]
+        public List<LessonInfoWithStudentsAndCommentsOutputModel> GetAllLessonsWithStudentsAndComments()
+        {
+            var dto = _lessonService.SelectAllLessons();
+            return _mapper.Map<List<LessonInfoWithStudentsAndCommentsOutputModel>>(dto);
+        }
 
         // api/lesson/{lessonId}/comment/{commentId}
         [HttpPost("{lessonId}/comment/{commentId}")]
+        [Description("Add a lesson's comment.")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
         public void AddLessonComment(int lessonId, int commentId)
```

Comment:
please rename to AddCommentToLesson
and change to (int lessonId, CommentInputModel commentInputModel)